sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,311 @@
+
+        is_inlier[self.decision_function(X) <= self.threshold_] = -1
+
+        return is_inlier
+
+    def decision_function(self, X=None):
+        """Opposite of the Local Outlier Factor of X (as bigger is better).
+
+        The local outlier factor (LOF) of a sample captures its
+        supposed `degree of abnormality'.
+        It is the average of the ratio of the local reachability density of
+        a sample and those of its k-nearest neighbors.
+
+        Parameters
+        ----------
+
+        X : array-like, shape (n_samples, n_features), optional
+            The query sample or samples to compute the Local Outlier Factor
+            w.r.t. the training samples. If None, makes prediction on the
+            training data without considering them as their own neighbors.
+
+        Returns
+        -------
+        lof_scores : array of shape (n_samples,)
+            The Local Outlier Factor of each input samples. The lower,
+            the more abnormal.
+        """
+        if X is not None:
+            X = check_array(X, accept_sparse='csr')
+
+            return -self._local_outlier_factor(X)  # as bigger is better
+
+        # if X is None, make prediction on X=X_train without considered each
+        # query point as its own neighbor.
+        else:
+            if hasattr(self, 'decision_function_train_'):
+                return self.decision_function_train_
+            else:
+                return -self._local_outlier_factor()  # as bigger is better
+
+    def _k_distance(self, X=None):
+        """
+        Compute the k_distance and the neighborhood of query samples wrt
+        training samples self._fit_X.
+        If X=None, neighbors of each sample self._fit_X are returned.
+        In this case, the query point is not considered its own neighbor.
+        """
+        distances, neighbors_indices = self.kneighbors(
+            X=X, n_neighbors=self.n_neighbors)
+        k_dist = distances[:, self.n_neighbors - 1]
+
+        return k_dist, neighbors_indices
+
+    def _local_reachability_density(self, X):
+        """The local reachability density (LRD)
+
+        The LRD of a sample is the inverse of the average reachability
+        distance of its k-nearest neighbors.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The samples to compute the LRD w.r.t. self._fit_X
+            If None, compute the LRD of self._fit_X w.r.t. itself.
+            (In this case samples are not considered in their own
+            neighborhood)
+
+        Returns
+        -------
+        local_reachability_density : array, shape (n_samples,)
+            The local reachability density of each sample.
+        """
+        neighbors_indices = (self.neighbors_indices_fit_X_ if X is None
+                             else self._k_distance(X)[1])
+
+        if X is None:
+            X = self._fit_X
+
+        n_samples = X.shape[0]
+
```

Comment:
Here you can add `n_jobs = _get_n_jobs(self.n_jobs)` and replace  `n_jobs=n_jobs` in line 240, 243